src/parquet/encodings/plain-encoding.h

```diff
@@ -163,19 +163,41 @@ class PlainEncoder : public Encoder<DType> {
 
   explicit PlainEncoder(
       const ColumnDescriptor* descr, MemoryAllocator* allocator = default_allocator())
-      : Encoder<DType>(descr, Encoding::PLAIN, allocator) {}
+      : Encoder<DType>(descr, Encoding::PLAIN, allocator),
+        values_sink_(new InMemoryOutputStream(IN_MEMORY_DEFAULT_CAPACITY, allocator)) {}
 
-  void Encode(const T* src, int num_values, OutputStream* dst) override;
+  std::shared_ptr<Buffer> FlushValues() override;
+  void Put(const T* src, int num_values) override;
+  void Encode(const T* src, int num_values, OutputStream* dst);
+
+ protected:
+  std::shared_ptr<InMemoryOutputStream> values_sink_;
```

Comment:
I guess in the current implementation this is not an issue since we are creating encoders per each ColumnWriter. But I am not sure if coupling the value buffer pointer with the encoders is a correct approach.